### ISSUE TITLE
Difference between `log_probs` and `probs` when set `save_probs`

### ISSUE THREAD

**Issue by songmingl** [dauparas/ProteinMPNN#123]
Does anyone know the difference between `log_probs` and `probs` in the npz file when set `--save_probs 1`?

According to the code [here](https://github.com/dauparas/ProteinMPNN/blob/main/protein_mpnn_run.py#L328) and [here](https://github.com/dauparas/ProteinMPNN/blob/main/protein_mpnn_run.py#L334), it seems that `probs` is the probability used to generate sequence while `log_probs` is the log-probs of the generated sequence in one forward pass. Is that right?

**Comment by fourier20laplace** [comment]
Hi! I’m curious whether you have figured out or understood the problem. Could you share any updates or insights?
Thanks in advance!

**Comment by songmingl** [comment]
`log_probs` is the log softmax results of logits: see codes [here](https://github.com/dauparas/ProteinMPNN/blob/main/protein_mpnn_utils.py#L1098-L1099), while `probs` is the softmax results of logits, adjusted for temperature and with the probability of omitted amino acids forced to 0 [(codes here)](https://github.com/dauparas/ProteinMPNN/blob/main/protein_mpnn_utils.py#L1165-L1179)

**Comment by fourier20laplace** [comment]
Thanks! By the way, I found that if I set the temperature to 1, the log_probs is approximately equal to the log of the probs (probabilities). This really confused me for a long time: why does the code calculate the probs using a forward pass, even though the sample function returns a log_prob? This might be because the author wanted to sample with a greedy policy but perform the forward pass in the normal way.
